sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java

```diff
@@ -98,29 +100,39 @@ private void configureChannelPipelineHandlers() {
         Objects.requireNonNull(request.httpMethod());
         Objects.requireNonNull(request.uri());
         Objects.requireNonNull(this.httpClientConfig);
-        if(request.getReactorNettyRequestRecord() == null) {
+        if(request.reactorNettyRequestRecord() == null) {
             ReactorNettyRequestRecord reactorNettyRequestRecord = new ReactorNettyRequestRecord();
             reactorNettyRequestRecord.setTimeCreated(Instant.now());
-            request.setReactorNettyRequestRecord(reactorNettyRequestRecord);
+            request.withReactorNettyRequestRecord(reactorNettyRequestRecord);
         }
 
         final AtomicReference<ReactorNettyHttpResponse> responseReference = new AtomicReference<>();
 
+        final Duration responseTimeout;
+        if (OperationType.QueryPlan.equals(request.operationType())) {
+            responseTimeout = Duration.ofSeconds(this.httpClientConfig.getConfigs().getQueryPlanResponseTimeoutInSeconds());
+        } else if (OperationType.AddressRefresh.equals(request.operationType())) {
+            responseTimeout = Duration.ofSeconds(this.httpClientConfig.getConfigs().getAddressRefreshResponseTimeoutInSeconds());
+        } else {
+            responseTimeout = Duration.ofSeconds(this.httpClientConfig.getConfigs().getDirectHttpsResponseTimeoutInSeconds());
+        }
```

Comment:
These is business logic. HttpClient should only be doing http network layer handling not business logic.

If we need to set a different response timeout per business logic operation then the caller of `HttpClient.Send` should pass the timeout as an override.

adding an overloaded method `HttpClient.send(httpRequest, responseTimeout)` and keeping the business logic out of the HttpClient to me seems a better approach.


That way you don't have to add operationType to the HttpRequest.

Reply:
@moderakh  - that is good point, we can update it. In fact, we don't even need overloaded API, we can just update the same API since `responseTimeout` is a must to have value now.

Reply:
This is done too, resolving the conversation.